### PR TITLE
Compatibility Lizmap 3.7

### PR DESCRIPTION
**Pull request by nworr**
+ Fix jelix call (iniModifier, configPath (break compatibilty with LCW 3.4/3.5 ?)
+ Fix js bug when project projection is 4326

untested with database as altitude provider (only IGN)

Update will be needed to use the IGN geoplateforme URL

### CODE REVIEW

File: altiProfil/lib/AltiConfig.php [hunk 1]
```diff
@@ -8,7 +8,7 @@ class AltiConfig
 
     function __construct()
     {
-        $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
```
Comment (laurentj) [inline altiProfil/lib/AltiConfig.php:11]: To keep compatibility with lizmap 3.5, we could use such code:

```
if (method_exists('jApp', 'varConfigPath')) {
       $altiProfilConfigFile = \jApp::varConfigPath('altiProfil.ini.php');
}
else {
      $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
}
```

File: altiProfilAdmin/controllers/config.classic.php [hunk 2]
```diff
@@ -24,8 +24,8 @@ class configCtrl extends jController {
 
     function __construct( $request ) {
         parent::__construct( $request );
-        $monfichier = jApp::configPath('altiProfil.ini.php');
```
Comment (laurentj) [inline altiProfilAdmin/controllers/config.classic.php:27]: Same proposal here with `if (method_exists('jApp', 'varConfigPath'))`

**Review by laurentj — CHANGES_REQUESTED** [review]

File: altiProfil/lib/AltiConfig.php [hunk 3]
```diff
@@ -8,7 +8,12 @@ class AltiConfig
...
+        if (method_exists('jApp', 'varConfigPath')) {
+            $altiProfilConfigFile = \jApp::varConfigPath('altiProfil.ini.php');
+        } else {
+            $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
+        }
+        $altiProfilConfigFile = \jApp::varconfigPath('altiProfil.ini.php');
```
Comment (laurentj) [inline altiProfil/lib/AltiConfig.php:16]: this line should be deleted

File: altiProfilAdmin/controllers/config.classic.php [hunk 4]
```diff
@@ -24,8 +24,12 @@ class configCtrl extends jController {
...
-        $this->ini = new jIniFileModifier ($monfichier);
+        if (method_exists('jApp', 'varConfigPath')) {
+		    $monfichier = \jApp::varConfigPath('altiProfil.ini.php');
+        } else {
+		    $monfichier = \jApp::configPath('altiProfil.ini.php');
+        }
+        $this->ini = new \Jelix\IniFile\IniModifier($monfichier);
```
Comment (laurentj) [inline altiProfilAdmin/controllers/config.classic.php:32]: you should include also this change into the if.
`if (method_exists('jApp', 'varConfigPath'))  ->  \Jelix\IniFile\IniModifier else jIniFileModifier`

**Review by laurentj — CHANGES_REQUESTED** [review]

File: altiProfil/lib/AltiConfig.php [hunk 5]
```diff
@@ -8,7 +8,11 @@ class AltiConfig
 
     function __construct()
     {
-        $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
+        if (method_exists('jApp', 'varConfigPath')) {
+            $altiProfilConfigFile = \jApp::varConfigPath('altiProfil.ini.php');
+        } else {
+            $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
```
Comment (Gustry) [inline altiProfil/lib/AltiConfig.php:14]:
```suggestion
            // LWC >= 3.7
            $altiProfilConfigFile = \jApp::varConfigPath('altiProfil.ini.php');
        } else {
            $altiProfilConfigFile = \jApp::configPath('altiProfil.ini.php');
```

File: altiProfilAdmin/controllers/config.classic.php [hunk 6]
```diff
@@ -24,8 +24,13 @@ class configCtrl extends jController {
...
     function __construct( $request ) {
         parent::__construct( $request );
-        $monfichier = jApp::configPath('altiProfil.ini.php');
-        $this->ini = new jIniFileModifier ($monfichier);
+        if (method_exists('jApp', 'varConfigPath')) {
+		    $monfichier = \jApp::varConfigPath('altiProfil.ini.php');
+            $this->ini = new \Jelix\IniFile\IniModifier($monfichier);
+        } else {
+		    $monfichier = \jApp::configPath('altiProfil.ini.php');
+            $this->ini = new jIniFileModifier($monfichier);
```
Comment (Gustry) [inline altiProfilAdmin/controllers/config.classic.php:32]:
```suggestion
        if (method_exists('jApp', 'varConfigPath')) {
	    // LWC >= 3.7
	    $monfichier = \jApp::varConfigPath('altiProfil.ini.php');
            $this->ini = new \Jelix\IniFile\IniModifier($monfichier);
        } else {
	    $monfichier = \jApp::configPath('altiProfil.ini.php');
            $this->ini = new jIniFileModifier($monfichier);
```

Mixed tab ?

**Review by Gustry — COMMENTED** [review]
It helps to understand and it will be easier to clean in a few versions ;-)

It was mainly for the mixed tab/space at the beginning

File: altiProfilAdmin/controllers/config.classic.php (hunk 6 above)
Comment (nworr) [reply altiProfilAdmin/controllers/config.classic.php:32]: My IDE drives me crazy, unable to get a correct indentation without modifiying the whole file i gave up 😭 😄

**Review by laurentj — APPROVED** [review]

**Pull request merged**